docs/manual/src/internals/object_references.md

```diff
@@ -56,91 +56,30 @@ interface TodoList {
-) -> *const std::os::raw::c_void /* *const TodoList */ {
-    uniffi::deps::ffi_support::call_with_output(err, || {
-        let _new = TodoList::new();
-        let _arc = std::sync::Arc::new(_new);
-        <std::sync::Arc<TodoList> as uniffi::FfiConverter>::lower(_arc)
-    })
-}
-```
-
-The UniFFI runtime implements lowering for object instances using `Arc::into_raw`:
-
-```rust
-unsafe impl<T: Sync + Send> FfiConverter for std::sync::Arc<T> {
-    type FfiType = *const std::os::raw::c_void;
-    fn lower(self) -> Self::FfiType {
-        std::sync::Arc::into_raw(self) as Self::FfiType
-    }
-}
-```
-
-which does the "arc to pointer" dance for us. Note that this has "leaked" the
-`Arc<>` reference out of Rusts ownership system and given it to the foreign-language code.
-The foreign-language code must pass that pointer back into Rust in order to free it,
-or our instance will leak.
-
-When invoking a method on the instance, the foreign-language code passes the
-raw pointer back to the Rust code, conceptually passing a "borrow" of the `Arc<>` to
-the Rust scaffolding. The Rust side turns it back into a cloned `Arc<>` which
-lives for the duration of the method call:
-
-```rust
-pub extern "C" fn todolist_12ba_TodoList_add_item(
-    ptr: *const std::os::raw::c_void,
-    todo: uniffi::RustBuffer,
-    err: &mut uniffi::deps::ffi_support::ExternError,
-) -> () {
-    uniffi::deps::ffi_support::call_with_result(err, || -> Result<_, TodoError> {
-        let _retval = TodoList::add_item(
-          &<std::sync::Arc<TodoList> as uniffi::FfiConverter>::try_lift(ptr).unwrap(),
-          <String as uniffi::FfiConverter>::try_lift(todo).unwrap())?,
-        )
-        Ok(_retval)
-    })
-}
-```
-
-The UniFFI runtime implements lifting for object instances using `Arc::from_raw`:
-
-```rust
-unsafe impl<T: Sync + Send> FfiConverter for std::sync::Arc<T> {
-    type FfiType = *const std::os::raw::c_void;
-    fn try_lift(v: Self::FfiType) -> Result<Self> {
-        let v = v as *const T;
-        // We musn't drop the `Arc<T>` that is owned by the foreign-language code.
-        let foreign_arc = std::mem::ManuallyDrop::new(unsafe { Self::from_raw(v) });
-        // Take a clone for our own use.
-        Ok(std::sync::Arc::clone(&*foreign_arc))
-    }
-```
-
-Notice that we take care to ensure the reference that is owned by the foreign-language
-code remains alive.
+On the Rust side of the generated bindings:
+ - The instance constructor will create an instance of the corresponding `TodoList` Rust struct
+ - The owned value is wrapped it in an `Arc<>`
+ - The `Arc<>` is lowered into the foreign code using `Arc::into_raw` and returned as an object pointer.
+
+This is the "arc to pointer" dance. Note that this has "leaked" the `Arc<>`
+reference out of Rusts ownership system and given it to the foreign-language
+code. The foreign-language code must pass that pointer back into Rust in order
+to free it, or our instance will leak.
```

Comment:
(On one hand I'm sad to see this example generated code be removed, but OTOH it's going to get even more out of date than it already is. I wonder if it might make sense to add a brief and vague note somewhere that people who want to deeply understand the generated code should look at using the `cargo expand` utility?)